File: BuildTest/BuildTest/ViewController.m
//
//  ViewController.m
//  BuildTest
//
//  Created by sunkai on 16/11/19.
//  Copyright © 2016年 imo. All rights reserved.
//

//#import <Foundation/Foundation.h>
//#import <UIKit/UIKit.h>
#import "ViewController.h"
#import "Masonry.h"

@interface ViewController ()<UITableViewDelegate,UITableViewDataSource>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
//
//    UIScrollView *scrollview = [[UIScrollView alloc] init];
//    [self.view addSubview:scrollview];
//    
//    //添加一个视图
//    UIView *view1 = [[UIView alloc] init];
//    view1.backgroundColor =  [UIColor blueColor];
//    
//    [scrollview addSubview:view1];
//    
//    //自动布局
//    [scrollview mas_makeConstraints:^(MASConstraintMaker *make) {
//        make.edges.equalTo(self.view);
//    }];
//    
//    [view1 mas_makeConstraints:^(MASConstraintMaker *make) {
//        make.edges.equalTo(scrollview);
//        make.size.mas_equalTo(CGSizeMake([UIScreen mainScreen].bounds.size.width, 1000));
//    }];
//    
//    //之后的全部布局控件，都是添加到view1上就行了额
    
    UITableView *table = [[UITableView alloc] initWithFrame:self.view.bounds style:UITableViewStylePlain];
    table.delegate = self;
    table.dataSource = self;
    [table registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cell"];
    [self.view addSubview:table];

}



-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return 100;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    cell.textLabel.text = @(indexPath.row).stringValue;
    return cell;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    if (indexPath.row == 99){
        UIViewController *vc = [[UIViewController alloc] init];
        vc.view.backgroundColor = [UIColor whiteColor];
        vc.hidesBottomBarWhenPushed = YES;
        self.navigationController.view.backgroundColor = [UIColor whiteColor];
        [self.navigationController pushViewController:vc animated:YES];
    }
}
    
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: RAC/RAC/ViewController.swift
//
//  ViewController.swift
//  RAC
//
//  Created by sunkai on 16/11/19.
//  Copyright © 2016年 imo. All rights reserved.
//

import UIKit
import Foundation


class ViewController: UIViewController {
    
    var viewModel : PersonListViewModel?
    var table : UITableView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.loadData()
    
       
    }
    
    
    func preparaTableview(){
        let tableview = UITableView.init(frame: self.view.bounds, style: .Plain)
        tableview.delegate = self
        tableview.dataSource = self
        table = tableview
        self.view.addSubview(table)
        tableview.registerClass(UITableViewCell.classForCoder(), forCellReuseIdentifier: "cell")
    }
    
    func  loadData(){
        if viewModel == nil {
            viewModel = PersonListViewModel()
            viewModel?.loadLists().subscribeNext({ (x) in
                Swift.print("next事件完成")
                 self.preparaTableview()
                
                }, completed: {
                    Swift.print("完成事件完成")
                    self.table.reloadData()
            })
        }
    }

}

extension ViewController : UITableViewDelegate,UITableViewDataSource{
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return (viewModel?.personList.count)!
    }
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell  = tableView.dequeueReusableCellWithIdentifier("cell", forIndexPath: indexPath)
        cell.textLabel?.text = viewModel?.personList[indexPath.row].name
        return cell
    }
    
}

